perfkitbenchmarker/object_storage_service.py

```diff
@@ -0,0 +1,250 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An interface to object storage services."""
+
+import abc
+import os
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+
+flags.DEFINE_string('object_storage_credential_file', None,
+                    'Directory of credential file.')
+flags.DEFINE_string('boto_file_location', None,
+                    'The location of the boto file.')
+
+FLAGS = flags.FLAGS
+
+DEFAULT_BOTO_LOCATION = '~/.boto'
+BOTO_LIB_VERSION = 'boto_lib_version'
+
+_OBJECT_STORAGE_REGISTRY = {}
+
+
+def GetObjectStorageClass(storage_name):
+  """Return the ObjectStorageService subclass corresponding to 'storage_name'"""
+
+  return _OBJECT_STORAGE_REGISTRY[storage_name]
+
+
+class AutoRegisterObjectStorageMeta(abc.ABCMeta):
+  def __init__(cls, name, bases, dct):
+    super(AutoRegisterObjectStorageMeta, cls).__init__(name, bases, dct)
+    if cls.STORAGE_NAME in _OBJECT_STORAGE_REGISTRY:
+      raise Exception('Duplicate storage implementations for name "%s"' %
+                      cls.STORAGE_NAME)
+    _OBJECT_STORAGE_REGISTRY[cls.STORAGE_NAME] = cls
+
+
+class ObjectStorageService(object):
+  __metaclass__ = AutoRegisterObjectStorageMeta
+
+  STORAGE_NAME = None
+
+  # Keeping the location in the service object is not very clean, but
+  # a nicer solution would be more complex, and we only use different
+  # locations in a very limited way. Also, true multi-location
+  # providers would require another abstraction for Azure service
+  # accounts, which would add more complexity.
+
+  # Service object lifecycle
+
+  def PrepareService(self, location):
+    """Get ready to use object storage.
+
+    This method should be called before any other method of this
+    class. Once it is called, all of this class' methods should work
+    with data in the given location.
+
+    Args:
+      location: where to place our data.
+    """
+
```

Comment:
I thought that looked weird, because we have that blank line when there's a real function body, but what you said seems to match the style guide. Done.